2009-10-26  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/unix/sysv/linux/s390/s390-32/getutmp.c: redefine 
	strong_alias instead of weak_alias before the include.
	Alias __getutmpx to __getutmp in this file.

diff --git a/sysdeps/unix/sysv/linux/s390/s390-32/getutmp.c b/sysdeps/unix/sysv/linux/s390/s390-32/getutmp.c
index 2a14e12..acb6270 100644
--- a/sysdeps/unix/sysv/linux/s390/s390-32/getutmp.c
+++ b/sysdeps/unix/sysv/linux/s390/s390-32/getutmp.c
@@ -26,13 +26,17 @@
 
 #include "utmp-compat.h"
 
-#undef weak_alias
-#define weak_alias(n,a)
+#undef strong_alias
+#define strong_alias(n,a)
 #define getutmp __getutmp
 #define getutmpx __getutmpx
 #include "sysdeps/gnu/getutmp.c"
 #undef getutmp
 #undef getutmpx
+#undef strong_alias
+#define strong_alias(name, aliasname) _strong_alias(name, aliasname)
+
+strong_alias (__getutmp, __getutmpx)
 
 default_symbol_version (__getutmp, getutmp, UTMP_COMPAT_BASE);
 default_symbol_version (__getutmpx, getutmpx, UTMP_COMPAT_BASE);
